src/Grpc.Core.Api/VersionInfo.cs

```diff
@@ -0,0 +1,46 @@
+#region Copyright notice and license
+
+// Copyright 2015 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#endregion
+
+namespace Grpc.Core
+{
+    /// <summary>
+    /// Provides info about current version of gRPC.
+    /// See https://codingforsmarties.wordpress.com/2016/01/21/how-to-version-assemblies-destined-for-nuget/
+    /// for rationale about assembly versioning.
+    /// </summary>
+    public static class VersionInfo
+    {
+        // TODO(jtattermusch): how to propagate versions from https://github.com/grpc/grpc-dotnet/blob/f238855fcb77cec01e87eb9cc08680c3420c132f/Directory.Build.props#L8-L10version into
```

Comment:
Still need to figure out how to turn .csproj properties into  string constants.
(in grpc/grpc we simply used codegen to generate contents of VersionInfo.cs)

Reply:
@JamesNK  any ideas how to do this?

Reply:
No. I know how to easily embed properties as assembly attributes, which can then be fetched at runtime. But these are const values so that won't work.

Reply:
Could write a source generator? https://docs.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview